types/alexa-sdk/index.d.ts

```diff
@@ -154,18 +251,682 @@ export interface OutputSpeech {
 }
 
 export interface Card {
-    type: "Simple" | "Standard" | "LinkAccount";
+    type: CardType;
     title?: string;
     content?: string;
     text?: string;
-    image?: Image;
-}
-
-export interface Image {
-    smallImageUrl: string;
-    largeImageUrl: string;
+    image?: CardImage;
 }
 
 export interface Reprompt {
     outputSpeech: OutputSpeech;
 }
+
+export interface ApiClientOptions {
+    hostname: string;
+    port: string;
+    path: string;
+    protocol: string;
+    headers: string;
+    method: string;
+}
+export interface ApiClientResponse {
+    statusCode: string;
+    statusText: string;
+    body: object;
+    headers: object;
+}
+/**
+ * Todo-ListItem class
+ * Refer https://developer.amazon.com/docs/custom-skills/access-the-alexa-shopping-and-to-do-lists.html
+ */
+export interface ListItemObject {
+    /**
+     * item id (String, limit 60 characters)
+     */
+    id: string;
+    /**
+     * item value (String, limit is 256 characters)
+     */
+    value: string;
+    /**
+     * item status
+     */
+    status?: ListItemObjectStatus;
+    /**
+     * item version (Positive integer | string)
+     */
+    version?: any;
+    /**
+     * created time (ISO 8601 time format with time zone)
+     */
+    createdTime: Date;
+    /**
+     * updated time (ISO 8601 time format with time zone)
+     */
+    updatedTime: Date;
+    /**
+     *  URL to retrieve the item (String)
+     */
+    href?: string;
+}
+/**
+ * Todo-List class
+ * Refer https://developer.amazon.com/docs/custom-skills/access-the-alexa-shopping-and-to-do-lists.html
+ */
+export interface ListObject {
+    /**
+     * list id (String)
+     */
+    listId: string;
+    /**
+     * list name (String)
+     */
+    name: string;
+    /**
+     * state
+     * "active" or "archived" (Enum)
+     */
+    state?: ListObjectState;
+    /**
+     * Possibly status of the list (or state?)
+     * Fetched from commit eebba0d at https://github.com/alexa/alexa-skills-kit-sdk-for-nodejs/
+     * File path: alexa-skills-kit-sdk-for-nodejs/lib/services/listManagementService.js
+     */
+    status?: string;
+    /**
+     * list version (long | string)
+     */
+    version?: any;
```

Comment:
Type information is in the comments and not the type annotations.

Reply:
It keeps throwing "no-redundant-jsdoc-2  JSDoc tag '@type' is redundant in TypeScript code." if we put type annotations. I have updated it with a type number.